docs/how_to_write_custom_backend.rst

```diff
@@ -0,0 +1,193 @@
+
+    def send_sms(self, number, message):
+        # Implement your service's SMS sending functionality
+        self.client.send_message({
+            'from': self._from,
+            'to': number,
+            'text': message,
+        })
+
+5. For sending bulk messages, you must override ``send_bulk_sms`` method of ``phone_verify.backends.base.BaseBackend`` class. It must have two positional parameters ``numbers`` and ``message`` respectively.
+
+.. code-block:: python
+
+    ...
+
+    def send_sms(self, number, message):
+        # Implement your service's SMS sending functionality
+        self.client.send_message({
+            'from': self._from,
+            'to': number,
+            'text': message,
+        })
+
+    def send_bulk_sms(self, numbers, message):
+        for number in numbers:
+            self.send_sms(self, number=number, message=message)
+
+How to create custom Sandbox Service?
+-------------------------------------
+
+The above steps will remain same if you wish to create a sandbox utility for your service. We'll create a new class with keeping above steps in mind. Apart from it, we will need to override a few more methods and tweak our ``__init__`` method a bit.
+
+1. Create a custom sandbox class for your service, ``NexmoSandboxBackend`` inherited from ``phone_verify.backends.base.BaseBackend``.
+
+2. The constructor for your sandbox environment can get the ``SANDBOX_TOKEN`` from settings as shown. This enables you to keep the token constant for testing purposes. You can then override ``send_sms`` and ``send_bulk_sms`` for the service as done in case of creating actual custom backend.
+
+.. code-block:: python
+
+    # Third Party Stuff
+    import nexmo
+    from phone_verify.backends.base import BaseBackend
+
+
+    class NexmoSandboxBackend(BaseBackend):
+
+        def __init__(self, **options):
+            super().__init__(**options)
+
+            # Lower case it just to be sure
+            options = {key.lower(): value for key, value in options.items()}
+            self._key = options.get("key", None)
+            self._secret = options.get("secret", None)
+            self._from = options.get("from", None)
+            self._token = options.get("sandbox_token", None)  # Fetch sandbox token for your service.
+
+            # Create a Nexmo Client object
+            self.client = nexmo.Client(key=self._key, secret=self._secret)
+
+        def send_sms(self, number, message):
+            # Implement your service's SMS sending functionality
+            self.client.send_message({
+                'from': self._from,
+                'to': number,
+                'text': message,
+            })
+
+        def send_bulk_sms(self, numbers, message):
+            for number in numbers:
+                self.send_sms(self, number=number, message=message)
+
+3. Override ``generate_security_code`` as defined below:
+
+.. code-block:: python
```

Comment:
It is not clean if `generate_security_code` and `validate_security_code` should be methods of the same class.

I'd suggest to have just one single code example instead of breaking in steps.